provider/cmd/pulumi-resource-nuage/nuage_provider/container_function.py

```diff
@@ -39,162 +46,253 @@ def docker_value(self) -> str:
+                                    "countType": "sinceImagePushed",
+                                    "countUnit": "days",
+                                    "countNumber": 30,
+                                },
+                                "action": {"type": "expire"},
+                            }
+                        ]
+                    }
+                ),
+                opts=pulumi.ResourceOptions(parent=repository),
+            )
+
+        architecture = Architecture[args.architecture]
+
+        extra_options = ["--platform", architecture.docker_value, "--quiet"]
+        if os.getenv("GITHUB_ACTIONS"):
+            # If we're running on a GitHub Actions runner, enable Caching API
+            extra_options += ["--cache-to=type=gha,mode=max", "--cache-from=type=gha"]
+
         if args.dockerfile:
             # Use specified dockerfile.
             build = docker.DockerBuild(
                 context=args.context or "./",
                 dockerfile=args.dockerfile,
-                extra_options=["--platform", architecture.docker_value, "--quiet"]
+                extra_options=extra_options,
             )
             image_ignore_changes = []
         else:
+            tmp = tempfile.NamedTemporaryFile(dir="./", delete=True)
             # Use default aws lambda docker image.
-            with open('Dockerfile.awslambda.generated','w') as f:
-                f.writelines([
-                    'FROM public.ecr.aws/lambda/provided:al2',
-                    '\n'
-                    'CMD [ "function.handler" ]'
-                ])
+            with open(tmp.name, "w") as f:
+                f.writelines(
+                    [
+                        "FROM public.ecr.aws/lambda/provided:al2",
+                        "\n" 'CMD [ "function.handler" ]',
+                    ]
+                )
 
             build = docker.DockerBuild(
                 context="./",
-                dockerfile="./Dockerfile.awslambda.generated",
-                extra_options=["--platform", architecture.docker_value, "--quiet"]
+                dockerfile=tmp.name,
+                extra_options=extra_options,
             )
             # Ignore changes on image_name if default docker image is used.
             image_ignore_changes = ["image_name"]
-        
+
         # Authenticate with ECR and get cridentals.
-        registry_id = repository.apply(lambda x: x.split(".")[0])
+        registry_id = repository.repository_url.apply(lambda x: x.split(".")[0])
         auth = aws.ecr.get_authorization_token(registry_id=registry_id)
-        
+
         # Build and Push docker Image.
         image = docker.Image(
-            name=f"{name}-image",
-            build = build,
-            image_name = repository,
+            name=f"{resource_name}-image",
+            build=build,
+            image_name=repository.repository_url.apply(
+                lambda url: f"{url}:{resource_name}"
```

Comment:
`resource_name` should only be used in the context of Pulumi

Since this is an actual name deployed to ECR, I'm guessing we should use `name`